polygeist/tools/cgeist/Test/Verification/sycl/illegal_methodop.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: clang++ -fsycl -fsycl-device-only -O0 -w -emit-mlir %s -o - | FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+// We should be generating the following error, but not the `sycl.id.get`
+// operation.
+
+// CHECK-LITERAL:  error: 'sycl.id.get' op result #0 must be 64-bit signless integer or memref of 64-bit signless integer values, but got 'f32'
```

Comment:
This error will be emitted in two cases:
1. The user defines a conflicting function in the `sycl` namespace, which is UB [by the spec](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:headers-and-namespaces)
2. We are not handling something defined in the runtime, so we should look into it.

I'd rather emit a warning or a debug trace, but `mlir` lacks the infrastructure for this. I think checking that we support the signature being used is more important.

A different solution would involve reimplementing the `verify` behavior and that's way out of scope given than `verify` is a one-liner that can save us of some undesired situations, e.g., a runtime implementing a deprecated signature of a function instead of the SYCL 2020 one.